hal/src/sercom/v2/uart/charsize.rs

```diff
@@ -0,0 +1,114 @@
+//! Character size definitions
+
+use crate::typelevel::Sealed;
+use num_traits::{AsPrimitive, PrimInt};
+
+/// Size of the SERCOM's `DATA` register
+#[cfg(any(feature = "samd11", feature = "samd21"))]
+pub type DataSize = u16;
+
+/// Size of the SERCOM's `DATA` register
+#[cfg(any(feature = "min-samd51g"))]
+pub type DataSize = u32;
```

Comment:
Seems like this belongs somewhere else. I'm not sure where though. Maybe the top level module? It doesn't affect the character size though.

Reply:
I realized a better name for this might be DataReg. I'm going to change it in the spi module.

I think DataSize sounds too related to CharSize. And in the spi module, I have a Size trait that could also be confused.

The intent is for this alias to represent the data register, so I think DataReg probably makes the most sense while also being unambiguous. Thoughts?

Reply:
Yup, that works!